apollo-federation/src/merge.rs

```diff
@@ -1259,3 +1350,127 @@ fn merge_directive(
         supergraph_directives.insert(directive.name.clone(), directive.clone());
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use apollo_compiler::Schema;
+    use insta::assert_snapshot;
+
+    use crate::merge::merge_federation_subgraphs;
+    use crate::schema::ValidFederationSchema;
+    use crate::ValidFederationSubgraph;
+    use crate::ValidFederationSubgraphs;
+
+    #[test]
+    fn test_steel_thread() {
+        let one_sdl =
+            include_str!("./sources/connect/expand/merge/connector_Query_users_0.graphql");
+        let two_sdl = include_str!("./sources/connect/expand/merge/connector_Query_user_0.graphql");
+        let three_sdl = include_str!("./sources/connect/expand/merge/connector_User_d_1.graphql");
+        let graphql_sdl = include_str!("./sources/connect/expand/merge/graphql.graphql");
+
+        let mut subgraphs = ValidFederationSubgraphs::new();
+        subgraphs
+            .add(ValidFederationSubgraph {
+                name: "connector_Query_users_0".to_string(),
+                url: "".to_string(),
+                schema: ValidFederationSchema::new(
+                    Schema::parse_and_validate(one_sdl, "./connector_Query_users_0.graphql")
+                        .unwrap(),
+                )
+                .unwrap(),
+            })
+            .unwrap();
+        subgraphs
+            .add(ValidFederationSubgraph {
+                name: "connector_Query_user_0".to_string(),
+                url: "".to_string(),
+                schema: ValidFederationSchema::new(
+                    Schema::parse_and_validate(two_sdl, "./connector_Query_user_0.graphql")
+                        .unwrap(),
+                )
+                .unwrap(),
+            })
+            .unwrap();
+        subgraphs
+            .add(ValidFederationSubgraph {
+                name: "connector_User_d_1".to_string(),
+                url: "".to_string(),
+                schema: ValidFederationSchema::new(
+                    Schema::parse_and_validate(three_sdl, "./connector_User_d_1.graphql").unwrap(),
+                )
+                .unwrap(),
+            })
+            .unwrap();
+        subgraphs
+            .add(ValidFederationSubgraph {
+                name: "graphql".to_string(),
+                url: "".to_string(),
+                schema: ValidFederationSchema::new(
+                    Schema::parse_and_validate(graphql_sdl, "./graphql.graphql").unwrap(),
+                )
+                .unwrap(),
+            })
+            .unwrap();
+
+        let result = merge_federation_subgraphs(subgraphs).unwrap();
+
+        let schema = result.schema.into_inner();
+        let validation = schema.clone().validate();
+        assert!(validation.is_ok(), "{:?}", validation);
+
+        assert_snapshot!(schema.serialize(), @r###"
```

Comment:
Why inline this here rather than having it in a snapshot file?

Reply:
yeah this could definitely be a file, no reason for it to be inline in this case